src/content/guides/ecommerce-website.mdx

```diff
@@ -5,27 +5,125 @@ nav:
+  Consumer->>CP: Click 'Pay with Centrapay' button
+  Note over CP: Open Centrapay Checkout popup
+  CP->>+MP: onClick() callback
+  activate MP
+  MP->>MS: Create Payment Request
+  MS->>CP: Create Payment Request
+  CP-->>MS: Payment Request created
+  MS-->>MP: Payment Request created
+  MP-->>-CP: Return Payment Request ID
+  Note over CP: Show Payment Request details
+  Consumer->>CP: Complete payment
+  Note over CP: Close Centrapay Checkout popup
+  CP->>+MP: onComplete() callback
+  opt Payment Request has status 'new'
+	MP->>MS: Void Payment Request
+  MS->>CP: Void Payment Request
+  CP-->>MS: Return Payment Request
+  MS-->>MP: Return Payment Request
+  end
+  MP-->>-CP: Return
+  Note over MP: Checkout complete ✅
+```
 
-### Optional Protocol Extensions
+- The `<script>` tag fetches the Centrapay Checkout SDK when your checkout page renders.
+- When the customer clicks the ‘Pay with Centrapay’ button, the `onClick` callback tells you to create a Payment Request.
+- The `onComplete` callback is triggered when the checkout process is complete.
 
-Our payment protocol supports several optional extensions. Please review the extensions below and determine which ones you need for your integration.
+## Know before you code
 
-- [Line Items](/guides/line-items)
-- [Requesting Pre Auth](/guides/requesting-pre-auth)
+Please get in touch with Centrapay to configure your business for eCommerce.
+Centrapay will provide you with the following resources to allow you to start creating [Payment Requests](/api/payment-requests).
 
-## Contact Us
+1. API Key.
+2. Merchant Config ID.
+
+> We strongly recommend Centrapay APIs are invoked from your backend where your API key is securely stored.
+
+### Access the Centrapay Checkout SDK
+
+The Centrapay Checkout SDK enables acceptance of Centrapay payments on your website.
+It handles displaying the Centrapay button and launching the Centrapay checkout.
+
+Production: `https://service.centrapay.com/ecommerce/pay-with-centrapay.js`
+
+## Popup Method Implementation
 
-Contact integrations@centrapay.com to get started with API keys.
+You can use the Popup Method to open the Centrapay Checkout in a new browser window.
+
+For desktop applications, the Centrapay Checkout window appears as an overlay on top of the merchant website.
+For mobile applications, the Centrapay Checkout opens in a new browser tab.
+
+<CodePanel title="Sample Integration" fullHeight="true">
+```html
+  <html>
+  <head>
+    <script src="https://service.centrapay.com/ecommerce/pay-with-centrapay.js"></script>
+  </head>
+  <body>
+    <div id="centrapay-button-container"></div>
+    <script type="text/javascript">
+      window.centrapay({
+        async onClick() {
+          // Create Payment Request
+          // Return Payment Request ID
+        },
+        async onComplete(data) {
+          if (data.paymmentRequest.status === 'new') {
+            // Void Payment Request
+          }
+        }
+      });
```

Comment:
Did you consider separating this into two functions i.e. `onSuccess` and `onFailure`, where the `onSuccess` callback gets called on a successfull payment and `onFailure` callback gets called for unsuccessfull payments.

Reply:
@Neeraj-Ghodla we could do that but it seems easier to implement one function. What benefit do your think that would give?

Reply:
I though by separating the integrator don't need to worry about the status of the payment request when those callbacks are called, but I think even with the functions spearated you'd still need to check the status in the `onFailure` case because you can't void expired requests.

Reply:
@Neeraj-Ghodla yes I think they would still need to check so will leave this as is. This page is experimental so we can always change it later if we see a need to.